src/app/Header/components/MenuRight/MenuRight.jsx

```diff
@@ -26,25 +27,36 @@ const SearchIcon = styled.div`
   left: -30px;
   top: 9px;
 `;
-const MenuRight = () => (
-  <Right>
-    <li>
-      <SearchBar>
-        <SearchInput type="text" />
-        <SearchIcon>
-          <Image
-            alt="SearchIcon"
-            src="/images/icons/searchIcon.svg"
-            layout="fill"
-            objectFit="contain"
-          />
-        </SearchIcon>
-      </SearchBar>
-    </li>
-    <li>
-      <SignIn size="lg">登陆</SignIn>
-    </li>
-  </Right>
-);
+
+const MenuRight = () => {
+  const [showLoginModal, setShowLoginModal] = useState(false);
+
+  const handleClick = (event) => {
+    event.preventDefault();
+    setShowLoginModal((preState) => !preState);
+  };
+
+  return (
+    <Right>
+      <li>
+        <SearchBar>
+          <SearchInput type="text" />
+          <SearchIcon>
+            <Image
+              alt="SearchIcon"
+              src="/images/icons/searchIcon.svg"
+              layout="fill"
+              objectFit="contain"
+            />
+          </SearchIcon>
+        </SearchBar>
+      </li>
+      <li>
+        <SignIn size="lg" onClick={handleClick}>登陆</SignIn>
+        {showLoginModal && <Login handleClick={handleClick} />}
```

Comment:
这个 props 应该叫 onClose 吧？
```suggestion
        {showLoginModal && <Login onClose={handleClick} />}
```